src/libos/crates/async-rt/src/wait/waiter.rs

```diff
@@ -40,6 +44,17 @@ impl Waiter {
         self.inner.wait()
     }
 
+    #[cfg(feature = "timeout")]
+    pub fn wait_timeout<'a, T: BorrowMut<Duration>>(
+        &'a self,
+        timeout: Option<&'a mut T>,
+    ) -> WrapperFuture<WaitFuture<'_>, Timeout<WaitFuture<'_>>> {
```

Comment:
I agree that using `impl Future<Output=Result<()>> + '_` as the return type.

About another issue,  if we combine these two functions, this will cause the user program to change as feature changes. I'm not sure if this is a good idea.

Reply:
> I agree that using `impl Future<Output=Result<()>> + '_` as the return type.

I didn't see the changes. You have a second thought?

> About another issue, if we combine these two functions, this will cause the user program to change as feature changes. I'm not sure if this is a good idea.

I agree. This is a valid point. Let's keep the two methods.

Reply:
Remove the `timeout` feature and make `timeout` as necessary instead of optional.
And we still keep two methods.